### PR TITLE
Fix ID mapping issue with deleted entries (TRM-27838)

### PR DESCRIPTION
**Pull request by LeonardoGonzales**

### CODE REVIEW

File: uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/repository/store/ResultsConfigTest.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package org.uniprot.api.uniprotkb.repository.store;
```
Comment (ahmadshadab) [inline uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/repository/store/ResultsConfigTest.java:2]: It seems this file has only formatting changes.

File: idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/impl/UniProtKBIdService.java [hunk 2]
```diff
@@ -56,15 +70,42 @@ public UniProtKBIdService(
         this.streamConfig = streamConfig;
         this.storeClient = storeClient;
         this.storeFetchRetryPolicy = storeFetchRetryPolicy;
+        this.solrClient = solrClient;
     }
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/impl/UniProtKBIdService.java:64]: I moved to a repository

File: idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/impl/UniProtKBIdService.java [hunk 3]
```diff
@@ -56,15 +70,42 @@ public UniProtKBIdService(
...
+        try {
+            SolrDocument solrDocument =
+                    solrClient.getById(SolrCollection.uniprot.toString(), accession);
+            if (solrDocument != null) {
+                UniProtDocument document =
+                        new DocumentObjectBinder().getBean(UniProtDocument.class, solrDocument);
+                if (document.active) {
+                    throw new ServiceException(
+                            "Error fetching deleted entry for id-mapping. accession:" + accession);
+                } else {
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/impl/UniProtKBIdService.java]: thanks, message improved

File: idmapping-rest/src/test/java/org/uniprot/api/idmapping/controller/UniProtKBIdMappingResultsControllerIT.java [hunk 4]
```diff
@@ -169,6 +180,27 @@ void testUniProtKBToUniProtKBMapping() throws Exception {
...
+                getJobOperation()
+                        .createAndPutJobInCache(
+                                UNIPROTKB_AC_ID_STR, UNIPROTKB_STR, "Q00001,I8FBX0");
+        ResultActions response =
+                mockMvc.perform(
+                        get(UNIPROTKB_ID_MAPPING_RESULT_PATH, job.getJobId())
+                                .header(ACCEPT, MediaType.APPLICATION_JSON));
+        // then
+        response.andDo(print())
+                .andExpect(status().is(HttpStatus.OK.value()))
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply idmapping-rest/src/test/java/org/uniprot/api/idmapping/controller/UniProtKBIdMappingResultsControllerIT.java:195]: thanks.. fixed it

File: idmapping-rest/src/test/java/org/uniprot/api/idmapping/controller/UniProtKBIdMappingResultsControllerIT.java [hunk 5]
```diff
@@ -312,4 +344,15 @@ void testIdMappingWithSplitQuerySuccess() throws Exception {
     protected String getDefaultSearchQuery() {
         return "FGF1"; // geneName
     }
+
+    private void saveInactiveEntries() throws IOException, SolrServerException {
+        UniProtDocument inactiveDoc = new UniProtDocument();
```
[the remark this answers is not in the document]
Comment (LeonardoGonzales) [reply idmapping-rest/src/test/java/org/uniprot/api/idmapping/controller/UniProtKBIdMappingResultsControllerIT.java:348]: updated it

**Pull request merged**